lib/src/over_react_redux/over_react_flux.dart

```diff
@@ -0,0 +1,599 @@
+    }
+  }
+  /*--end usage of cases--*/
+
+  if (areStatePropsEqual == null) {
+    const defaultAreStatePropsEqual = _shallowMapEquality;
+    const propHasher = CollectionLengthHasher();
+    bool areStatePropsEqualWrapper(TProps nextProps, TProps prevProps) {
+      final result = defaultAreStatePropsEqual(nextProps, prevProps);
+
+      assert(() {
+        prevProps.forEach((key, value) {
+          // If the value is the same instance, check if the instance has been mutated,
+          // causing its hash to be updated
+          if (identical(value, nextProps[key]) &&
+              propHasher.hasHashChanged(value)) {
+            window.console.warn(
+                'connect: The instance of the value mapped from store "$TStore" to prop "$key" was mutated directly, which will prevent updates from being detected.'
+                ' Instead of mutating datastructure instances within the store, overwrite them with modified copies.\n'
+                '\n  Good: `_items = [..._items, newItem]`'
+                '\n  Bad:  `_items.add(newItem)`'
+                '\n\nIf this is not possible, then either:'
+                '\n - update `areStatePropsEqual` to reflect this behavior: `areStatePropsEqual: (_, __) => false`'
+                '\n - set `pure: false`');
+          }
+        });
+
+        return true;
+      }());
+
+      return result;
+    }
+
+    areStatePropsEqual = areStatePropsEqualWrapper;
+  }
+
+  return connect(
+    mapStateToProps: mapStateToProps,
+    mapStateToPropsWithOwnProps: mapStateToPropsWithOwnProps,
+    mergeProps: mergeProps,
+    areOwnPropsEqual: areOwnPropsEqual,
+    areStatePropsEqual: areStatePropsEqual,
+    areMergedPropsEqual: areMergedPropsEqual,
+    context: context,
+    pure: pure,
+    forwardRef: forwardRef,
+    // Don't pass along the dispatcher as props by default
+    mapDispatchToProps: (_) => const {},
+    // This will always be false, since both arguments will also be the same store
+    areStatesEqual: (_, __) => false,
+  );
+}
+
+extension InfluxStoreExtension<S extends InfluxStoreMixin> on S {
+  /// Returns a [FluxToReduxAdapterStore] instance from the Flux store instance.
+  ///
+  /// This is meant to be a more succinct way to instantiate the adapter store.
+  FluxToReduxAdapterStore asReduxStore(dynamic actions,
+      {List<redux.Middleware> middleware}) {
+    if (_storeAdapterFor[this] != null) return _storeAdapterFor[this];
+
+    _storeAdapterFor[this] = FluxToReduxAdapterStore(this, actions, middleware: middleware);
+    return _storeAdapterFor[this];
+  }
+}
+
+extension FluxStoreExtension<S extends flux.Store> on S {
+  /// Returns a [ConnectFluxAdapterStore] instance from the Flux store instance.
+  ///
+  /// This is meant to be a more succinct way to instantiate the adapter store.
+  ConnectFluxAdapterStore asConnectFluxStore(dynamic actions,
+      {List<redux.Middleware<S>> middleware}) {
+    if (this is InfluxStoreMixin) {
+      throw ArgumentError(
```

Comment:
Can we use `ArgumentError.value` to provide an error message?

Reply:
The invalid value and the name of the parameter were a little difficult to define since they're both related to object instances rather than a function parameter, so I used what makes sense to me but let me know if it's still not ideal!